src/main/java/net/dv8tion/jda/api/interactions/components/Button.java

```diff
@@ -568,4 +568,40 @@ static Button of(@Nonnull ButtonStyle style, @Nonnull String idOrUrl, @Nonnull E
         Checks.notLonger(idOrUrl, 100, "Id");
         return new ButtonImpl(idOrUrl, "", style, false, emoji);
     }
+
+    /**
+     * Create an enabled button with the provided {@link ButtonStyle style}, URL or ID, label and emoji.
+     
+     * You can use {@link #asDisabled()} to disable it.
+     *
+     * <p>See {@link #link(String, String)} or {@link #primary(String, String)} for more details.
+     *
+     * @param  style
+     *         The button style
+     * @param  idOrUrl
+     *         Either the ID or URL for this button
+     * @param  label
+     *         The text to display on the button
+     * @param  emoji
+     *         The emoji to use as the button label
+     *
+     * @throws IllegalArgumentException
+     *         If any argument is empty or null, the label is longer than 80 characters, the id is longer than 100 characters, or the url is longer than 512 characters
+     *
+     * @return The button instance
+     */
+    @Nonnull
+    static Button of(@Nonnull ButtonStyle style, @Nonnull String idOrUrl, @Nonnull String label, @Nonnull Emoji emoji)
+    {
+        Checks.check(style != ButtonStyle.UNKNOWN, "Cannot make button with unknown style!");
+        Checks.notNull(style, "Style");
+        Checks.notNull(label, "Label");
+        Checks.notLonger(label, 80, "Label");
+        Checks.notNull(emoji, "Emoji");
+        if (style == ButtonStyle.LINK)
+            return link(idOrUrl, emoji).withLabel(label);
+        Checks.notEmpty(idOrUrl, "Id");
+        Checks.notLonger(idOrUrl, 100, "Id");
+        return new ButtonImpl(idOrUrl, label, style, false, emoji);
+    }
```

Comment:
```suggestion
    static Button of(@Nonnull ButtonStyle style, @Nonnull String idOrUrl, @Nullable String label, @Nullable Emoji emoji)
    {
        if (label != null)
            return of(style, idOrUrl, label).withEmoji(emoji);
        if (emoji != null)
            return of(style, idOrUrl, emoji);
        throw new IllegalArgumentException("Cannot build a button without a label and without an emoji. It must have at least one!");
    }
```

Reply:
What was the reasoning behind not making label nor emoji nullable in the other two `of` methods? I think, if I were to make it nullable, it should at least be consistent with the other methods that I will be calling internally (to fix the code repetition). 

Also, in the JavaDoc for the `of` method with just the Emoji, in the description, only `ButtonStyle` is linked and `Emoji` is not. Shall I also link it to the correct class in this method and my method?

Reply:
The reason is simply that you should not be able to make a button without emoji and without label. This is done by enforcing each method to accept one non-null parameter.